lib/Data/Database.php

```diff
@@ -297,14 +297,18 @@ public function exists($pasteid)
      * @param  array  $comment
      * @return bool
      */
-    public function createComment($pasteid, $parentid, $commentid, array $comment)
+    public function createComment($pasteid, $parentid, $commentid, array &$comment)
     {
         if (array_key_exists('data', $comment)) {
             $version = 1;
             $data    = $comment['data'];
         } else {
-            $version = 2;
-            $data    = Json::encode($comment);
+            try {
+                $version = 2;
+                $data    = Json::encode($comment);
+            } catch (Exception $e) {
+                return false;
```

Comment:
Would logging the exception as a warning make sense?

Reply:
The abstract createComment interface in AbstractData does not throw and returns bool. When I replicated the missing tests for Database, that exception made those fail, as they expected to detect the error via the return. We do also catch database related exceptions below.

Maybe we could revisit loggin caught exceptions in a future refactoring effort, now that we have such a mechanism.